src/xrCore/xrDebug.cpp

```diff
@@ -446,30 +446,34 @@ void xrDebug::GatherInfo(char* assertionInfo, size_t bufferSize, const ErrorLoca
     int nptrs = backtrace(array, 20);     // get void*'s for all entries on the stack
     char **strings = backtrace_symbols(array, nptrs);
 
-    if(strings)
+    if (strings)
     {
+        size_t demangledBufSize = 0;
+        char* demangledName = nullptr;
         for (size_t i = 0; i < nptrs; i++)
         {
             char* functionName = strings[i];
-            char* demangledName = nullptr;
+
             Dl_info info;
 
             if (dladdr(array[i], &info))
             {
-                int status = -1;
-                demangledName = abi::__cxa_demangle(info.dli_sname, nullptr,
-                    nullptr, &status);
-                if (status == 0)
+                if (info.dli_sname)
                 {
-                    functionName = demangledName;
+                    int status = -1;
+                    demangledName = abi::__cxa_demangle(info.dli_sname, demangledName, &demangledBufSize, &status);
+                    if (status == 0)
+                    {
+                        functionName = demangledName;
+                    }
                 }
             }
             Log(functionName);
-    #ifdef USE_OWN_ERROR_MESSAGE_WINDOW
+#ifdef USE_OWN_ERROR_MESSAGE_WINDOW
             buffer += xr_sprintf(buffer, bufferSize, "%s\n", functionName);
-    #endif // USE_OWN_ERROR_MESSAGE_WINDOW
-            ::free(demangledName);
+#endif // USE_OWN_ERROR_MESSAGE_WINDOW
         }
+        ::free(demangledName);
```

Comment:
Why outside of for loop?

Reply:
Because `abi::__cxa_demangle()` can reuse buffer and realloc it if needed. Less allocations == more performance. We can now crash faster.

Reply:
> We can now crash faster.

Got you. 😎